### PR TITLE
[SecurityBundle] FirewallConfig's user_checker should be mandatory

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | master
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/20404#issuecomment-258395500
| License       | MIT

### CODE REVIEW

**Comment by xabbuh** [comment]
👍

**Comment by chalasr** [comment]
Rebased after #20404

File: src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php [hunk 1]
```diff
@@ -18,44 +18,43 @@ class FirewallConfig
...
         $this->stateless = $stateless;
         $this->provider = $provider;
         $this->context = $context;
         $this->entryPoint = $entryPoint;
         $this->accessDeniedHandler = $accessDeniedHandler;
         $this->accessDeniedUrl = $accessDeniedUrl;
-        $this->userChecker = $userChecker;
         $this->listeners = $listeners;
     }
-
```
Comment (fabpot) [inline src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php]: should be reverted

Comment (chalasr) [inline src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php]: done

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**